File: src/main/java/ro/nicolaemariusghergu/queryit/dto/ManufacturerDto.java
package ro.nicolaemariusghergu.queryit.dto;

import lombok.*;

import java.util.Objects;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class ManufacturerDto {

    private Long id;

    private String name;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ManufacturerDto that = (ManufacturerDto) o;
        return id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/java/ro/nicolaemariusghergu/queryit/dto/ManufacturerDto.kt
package ro.nicolaemariusghergu.queryit.dto

import lombok.*
import java.util.*

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
class ManufacturerDto {
    private val id: Long? = null
    private val name: String? = null
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ManufacturerDto?
        return id == that.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }
}